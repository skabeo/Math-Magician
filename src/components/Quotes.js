import React, { useEffect, useState } from 'react';
import './styles.css';

const Quotes = () => {
  const [quotes, setQuotes] = useState('');
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  const API_KEY = '6Kt7Ap0M4voSDFXWsXfPFg==1VLMRvpruA8bmdms';
  const URL = 'https://api.api-ninjas.com/v1/quotes';

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        const response = await fetch(URL, {
          headers: {
            'X-Api-Key': API_KEY,
          },
        });
        const data = await response.json();
        setQuotes(data[0]);
        setLoading(false);
      } catch (error) {
        setError('Error Fetching quote');
        setLoading(false);
      }
    };
    fetchQuote();
  }, []);

  return (
    <div className="quotes-container">
      {loading && <p className="loading" />}
      {!loading && !error && (
        <div>
          <p className="italic">
            &quot;
            {quotes.quote}
            &quot;
          </p>
          <p>
            -
            {quotes.author}
          </p>
        </div>
      )}
      {!loading && error && <p>{error}</p>}
    </div>
  );
};

export default Quotes;
