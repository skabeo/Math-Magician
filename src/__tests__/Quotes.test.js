import React from 'react';
import { render } from '@testing-library/react';
import Quotes from '../components/Quotes';

describe('Quotes', () => {
  it('should render correctly', () => {
    const { container } = render(<Quotes />);
    expect(container).toMatchSnapshot();
  });
});
