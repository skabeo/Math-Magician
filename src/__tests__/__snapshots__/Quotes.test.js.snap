// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes should render correctly 1`] = `
<div>
  <div
    class="quotes-container"
  >
    <p
      class="loading"
    />
  </div>
</div>
`;
